tests/integration/test_task_pool.py

```diff
@@ -1243,6 +1245,147 @@ async def test_detect_incomplete_tasks(
             # ensure that it is correctly identified as incomplete
             assert itask.state.outputs.get_incomplete()
             assert itask.state.outputs.is_incomplete()
-            assert log_filter(log, contains=f"[{itask}] did not complete required outputs:")
+            assert log_filter(
+                log, contains=f"[{itask}] did not complete required outputs:")
             # the task should not have been removed
             assert itask in schd.pool.get_tasks()
+
+
+@pytest.mark.parametrize('compat_mode', ['compat-mode', 'normal-mode'])
+@pytest.mark.parametrize('cycling_mode', ['integer', 'datetime'])
+async def test_compute_runahead(
+    cycling_mode,
+    compat_mode,
+    flow,
+    scheduler,
+    start,
+    monkeypatch,
+):
+    """Test the calculation of the runahead limit.
+
+    This test ensures that:
+    * Runahead tasks are excluded from computations
+      see https://github.com/cylc/cylc-flow/issues/5825
+    * Tasks are initiated with the correct is_runahead status on statup.
+    * Behaviour in compat/regular modes is same unless failed tasks are present
+    * Behaviour is the same for integer/datetime cycling modes.
+
+    """
+    if cycling_mode == 'integer':
+        config = {
+            'scheduler': {
+                'allow implicit tasks': 'True',
+            },
+            'scheduling': {
+                'initial cycle point': '1',
+                'cycling mode': 'integer',
+                'runahead limit': 'P3',
+                'graph': {
+                    'P1': 'a'
+                },
+            }
+        }
+        point = lambda point: IntegerPoint(str(int(point)))
```

Comment:
[Flake8:E371](https://www.flake8rules.com/rules/E731.html) is not happy with this. I'm not sure how important it is in this context, but consider...
```suggestion
        def(point):
            return IntegerPoint(str(int(point)))
```

Reply:
Flake8 won't be happy with that inline function either.

IMO it's ok, especially for a test.

Reply:
I'll leave it as-is then.

Reply:
> Flake8 won't be happy with that inline function either.

You sure?

```console
> echo 'def point(point): return 42'> foo.py
> flake8 foo.py
...
```